arviz/data/io_pystan.py

```diff
@@ -67,19 +68,32 @@ def sample_stats_to_xarray(self):
         """Extract sample_stats from posterior."""
         posterior = self.posterior
 
-        # copy dims and coords
-        dims = deepcopy(self.dims) if self.dims is not None else {}
-        coords = deepcopy(self.coords) if self.coords is not None else {}
+        data = get_sample_stats(posterior)
 
-        # log_likelihood
-        log_likelihood = self.log_likelihood
-        if log_likelihood is not None:
-            if isinstance(log_likelihood, str) and log_likelihood in dims:
-                dims["log_likelihood"] = dims.pop(log_likelihood)
+        # lp__
+        stat_lp = get_draws(posterior, variables="lp__")
+        data["lp"] = stat_lp["lp__"]
 
-        data = get_sample_stats(posterior, log_likelihood)
+        return dict_to_dataset(data, library=self.pystan, coords=self.coords, dims=self.dims)
 
-        return dict_to_dataset(data, library=self.pystan, coords=coords, dims=dims)
+    @requires("posterior")
+    @requires("log_likelihood")
+    def log_likelihood_to_xarray(self):
+        """Store log_likelihood data in log_likelihood group."""
+        fit = self.posterior
+
+        # log_likelihood values
+        log_likelihood = self.log_likelihood
+        if isinstance(log_likelihood, str):
+            log_likelihood = [log_likelihood]
+        if isinstance(log_likelihood, (list, tuple)):
+            log_likelihood = {name: name for name in log_likelihood}
+        data = {
+            obs_var_name: get_draws(fit, variables=log_like_name)[log_like_name]
+            for obs_var_name, log_like_name in log_likelihood.items()
+        }

```

Comment:
I don't know if it really matters, but we could call  `get_draws` "once"

```
data = {
            obs_var_name: get_draws(fit, variables=log_like_name)[log_like_name]
            for obs_var_name, log_like_name in log_likelihood.items()
        }
```

---> 

```
log_likelihood_draws = get_draws(fit, variables=list(log_likelihood.keys())
data = {
            obs_var_name: log_likelihood_draws [log_like_name]
            for obs_var_name, log_like_name in log_likelihood.items()
        }
```

?